ksql-engine/src/test/java/io/confluent/ksql/integration/EndToEndIntegrationTest.java

```diff
@@ -144,18 +144,31 @@ public void shouldRetainSelectedColumnsInPartitionBy() throws Exception {
     queryMetadata.getKafkaStreams().start();
 
     BlockingQueue<KeyValue<String, GenericRow>> rowQueue = queryMetadata.getRowQueue();
-    KeyValue<String, GenericRow> nextRow = rowQueue.poll();
-    if (nextRow != null) {
-      List<Object> columns = nextRow.value.getColumns();
-      assertEquals(5, columns.size());
-      String pageid = columns.get(3).toString();
-      assertEquals(5, pageid.length());
-      assertEquals("PAGE_", pageid.substring(0, 5));
-
-      String userid = columns.get(4).toString();
-      assertEquals(5, userid.length());
-      assertEquals("USER_", userid.substring(0, 5));
-    }
+    TestUtils.waitForCondition(() -> {
```

Comment:
If you're using `TestUtils.waitforCondition` then you need to change the passed in lambda to only do a single poll call, returning false if no rows are consumed. `waitForCondition` will then call the lambda again...